### PR TITLE
docs(caching): improve Cube Store documentation

### PR DESCRIPTION
**Pull request by hassankhan**
Story details: https://app.clubhouse.io/cube-dev/story/469

### CODE REVIEW

**Review by rpaik — APPROVED** [review]

File: docs/content/Caching/Using-Pre-Aggregations.md [hunk 1]
```diff
@@ -319,17 +375,50 @@ variables; worker nodes **must** specify the `CUBESTORE_WORKER_PORT` and
...
+      - CUBESTORE_META_ADDR=localhost:9999
+  cube:
+    image: cubejs/cube:latest
+    ports:
+      - 4000:4000
+    env_file: .env
+    depends_on:
+      - cubestore
+    links:
+      - cubestore
```
Comment (igorlukanin) [inline docs/content/Caching/Using-Pre-Aggregations.md:403]: @hassankhan This is a copy-paste typo, right? Should not be just `cubestore`, I suppose?

Comment (hassankhan) [reply docs/content/Caching/Using-Pre-Aggregations.md]: Indeed it was 😀  fixed in the last push

File: docs/content/Caching/Using-Pre-Aggregations.md [hunk 2]
```diff
@@ -319,17 +377,58 @@ variables; worker nodes **must** specify the `CUBESTORE_WORKER_PORT` and
...
+    image: cubejs/cubestore:latest
+    environment:
+      - CUBESTORE_SERVER_NAME=localhost
+      - CUBESTORE_WORKER_PORT=9001
+      - CUBESTORE_META_ADDR=localhost:9999
+  cubestore_worker_2:
+    image: cubejs/cubestore:latest
+    environment:
+      - CUBESTORE_SERVER_NAME=localhost
+      - CUBESTORE_WORKER_PORT=9002
```
Comment (ilya-biryukov) [inline docs/content/Caching/Using-Pre-Aggregations.md:401]: I suggest using the same port on both workers.

File: docs/content/Caching/Using-Pre-Aggregations.md [hunk 3]
```diff
@@ -319,17 +377,58 @@ variables; worker nodes **must** specify the `CUBESTORE_WORKER_PORT` and
...
+  cube:
+    image: cubejs/cube:latest
+    ports:
+      - 4000:4000
+    environment:
+      - CUBEJS_CUBESTORE_HOST=localhost
+      - CUBEJS_CUBESTORE_PORT=9999
+    depends_on:
+      - cubestore_router
+    links:
```
Comment (ilya-biryukov) [inline docs/content/Caching/Using-Pre-Aggregations.md:412]: Let's remote the `links` section altogether.
Containers are linked together by default, this only seems to complicate things.

File: docs/content/Caching/Using-Pre-Aggregations.md [hunk 4]
```diff
@@ -319,17 +377,58 @@ variables; worker nodes **must** specify the `CUBESTORE_WORKER_PORT` and
...
 variables can be found [in the Environment Variables reference][ref-config-env].
 
+A sample Docker Compose stack setting this up might look like:
+
+```yaml
+version: '2.2'
+services:
+  cubestore_router:
+    image: cubejs/cubestore:latest
+    environment:
```
Comment (ilya-biryukov) [inline docs/content/Caching/Using-Pre-Aggregations.md:387]: We should expose relevant ports: 
```yaml
expose:
  - 9999
  - 3306 # This exposes MySQL endpoint.
  - 3030 # This exposes the HTTP endpoint for CubeJS.
```

Similarly, workers should expose their ports `9001`.

**Review by ilya-biryukov — CHANGES_REQUESTED** [review]
Mostly comments about the compose file from my side.
Other than that LG.

**Comment by igorlukanin** [comment]
@hassankhan Here we say which pre-aggregation storage layer is the default one: https://cube.dev/docs/deployment/production-checklist#set-up-pre-aggregations-storage. Do you think it should be updated to Cube Store as well?

**Comment by igorlukanin** [comment]
@hassankhan @ilya-biryukov Also, should we specify what is the minimum viable count of worker nodes and recommended profiles in AWS and GCP?

File: docs/content/Caching/Using-Pre-Aggregations.md [hunk 5]
```diff
@@ -299,12 +299,70 @@ folder in the project root during development.
...
+| Cube Store can further be configured via environment variables. To see a
+| complete reference, please consult the [Cube Store section of the Environment
+| Variables reference][ref-config-env].
+<!-- prettier-ignore-end -->
+
+Next, run Cube.js and tell it to connect to Cube Store running on `localhost` on
+port `3030`:
+
+```bash
+docker run -p 4000:4000 -e CUBEJS_CUBESTORE_HOST=localhost -e CUBEJS_CUBESTORE_PORT=3030 -v ~/cube-conf:/cube/conf/ cubejs/cube
```
Comment (keydunov) [inline docs/content/Caching/Using-Pre-Aggregations.md:323]: @hassankhan is it required to specify the `CUBEJS_CUBESTORE_PORT`? I'm leaning towards to keep as mich clear as possible and then link to the reference for additional options and overrides.

File: docs/content/Caching/Using-Pre-Aggregations.md (hunk 4 above)
Comment (keydunov) [reply docs/content/Caching/Using-Pre-Aggregations.md:388]: @ilya-biryukov @paveltiunov Could you please expand on who is a client for the "MySQL endpoint"? It seems Cube.js communicates with Cube Store via HTTP on 3030 port.

File: docs/content/Caching/Using-Pre-Aggregations.md [hunk 6]
```diff
@@ -319,17 +377,74 @@ variables; worker nodes **must** specify the `CUBESTORE_WORKER_PORT` and
...
+```
+
 ### Storage
 
 <!-- prettier-ignore-start -->
 [[warning | ]]
 | Cube Store can only use one type of remote storage at runtime.
 <!-- prettier-ignore-end -->
 
+The storage layer in Cube Store is built from the ground-up to take full
```
Comment (keydunov) [inline docs/content/Caching/Using-Pre-Aggregations.md:441]: @hassankhan I'm not sure it is technically correct. I'd probably rephrase it and share that Cube Store stores data parquet and it preferably to use storage like S3.

File: docs/content/Caching/Using-Pre-Aggregations.md [hunk 7]
```diff
@@ -319,17 +377,74 @@ variables; worker nodes **must** specify the `CUBESTORE_WORKER_PORT` and
...
+advantage of highly-available object storage. This improves performance
+massively.
+
 Cube Store requires read/write access to file storage in order to persist
-pre-aggregations as well as maintain its' own internal state. Currently, Cube
-Store can use a local path on the machine; alternatively both AWS S3 and Google
-Cloud are supported as highly-available, cloud storage.
+pre-aggregations as well as maintain its internal state. Cube Store can use both
+AWS S3 and Google Cloud, or if desired, a local path on the server.
```
Comment (keydunov) [inline docs/content/Caching/Using-Pre-Aggregations.md:447]: @hassankhan Can we do a quick example of how to use AWS S3? cc @ilya-biryukov

File: docs/content/Caching/Using-Pre-Aggregations.md [hunk 8]
```diff
@@ -299,12 +299,70 @@ folder in the project root during development.
...
+| Cube Store can further be configured via environment variables. To see a
+| complete reference, please consult the [Cube Store section of the Environment
+| Variables reference][ref-config-env].
+<!-- prettier-ignore-end -->
+
+Next, run Cube.js and tell it to connect to Cube Store running on `localhost`
+(on the default port `3030`):
+
+```bash
+docker run -p 4000:4000 -e CUBEJS_CUBESTORE_HOST=localhost -v ~/cube-conf:/cube/conf cubejs/cube
```
Comment (keydunov) [inline docs/content/Caching/Using-Pre-Aggregations.md:323]: @hassankhan maybe

```
docker run -p 4000:4000 \
    -e CUBEJS_CUBESTORE_HOST=localhost \ 
    -v ~/cube-conf:/cube/conf \ 
    cubejs/cube
```

File: docs/content/Caching/Using-Pre-Aggregations.md (hunk 4 above)
Comment (keydunov) [reply docs/content/Caching/Using-Pre-Aggregations.md:388]: @hassankhan let's remove any mention of 3306 port.

**Pull request merged**